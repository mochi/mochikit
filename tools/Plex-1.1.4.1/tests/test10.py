# Test traditional regular expression syntax.

import Test

from Plex.Traditional import re
from Plex.Errors import PlexError
from Plex import Seq, AnyBut

def test_err(s):
  try:
    print re(s)
  except PlexError, e:
    print e

print re("")
print re("a")
print re("[a]")
print re("[ab]")
print re("[abc]")
print re("[a-c]")
print re("[a-cd]")
print re("[a-cg-i]")
print re("[^a]")
print re("[^a-cg-i]")
print re("[-]")
print re("[-abc]")
print re("[abc-]")
print re("[]]")
print re("[]-]")
print re("[^-]")
print re("[^-abc]")
print re("[^abc-]")
print re("[^]]")
print re("[^]-]")
print re("a*")
print re("a+")
print re("a?")
print re("a*+?")
print re("ab")
print re("a|b")
print re("abcde")
print re("a|b|c|d|e")
print re("abc|def|ghi")
print re("abc(def|ghi)")
print re("ab\(c\[de")
print re("^abc$")
print str(re(".")) == str(Seq(AnyBut('\n')))
test_err("abc(de")
test_err("abc[de")
test_err("abc)de")


