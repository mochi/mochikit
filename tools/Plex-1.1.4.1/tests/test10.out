Seq()
Seq(Char('a'))
Seq(Any('a'))
Seq(Any('ab'))
Seq(Any('abc'))
Seq(Any('abc'))
Seq(Any('abcd'))
Seq(Any('abcghi'))
Seq(AnyBut('a'))
Seq(AnyBut('abcghi'))
Seq(Any('-'))
Seq(Any('-abc'))
Seq(Any('abc-'))
Seq(Any(']'))
Seq(Any(']-'))
Seq(AnyBut('-'))
Seq(AnyBut('-abc'))
Seq(AnyBut('abc-'))
Seq(AnyBut(']'))
Seq(AnyBut(']-'))
Seq(Rep(Char('a')))
Seq(Rep1(Char('a')))
Seq(Opt(Char('a')))
Seq(Opt(Rep1(Rep(Char('a')))))
Seq(Char('a'),Char('b'))
Alt(Seq(Char('a')),Seq(Char('b')))
Seq(Char('a'),Char('b'),Char('c'),Char('d'),Char('e'))
Alt(Seq(Char('a')),Seq(Char('b')),Seq(Char('c')),Seq(Char('d')),Seq(Char('e')))
Alt(Seq(Char('a'),Char('b'),Char('c')),Seq(Char('d'),Char('e'),Char('f')),Seq(Char('g'),Char('h'),Char('i')))
Seq(Char('a'),Char('b'),Char('c'),Alt(Seq(Char('d'),Char('e'),Char('f')),Seq(Char('g'),Char('h'),Char('i'))))
Seq(Char('a'),Char('b'),Char('('),Char('c'),Char('['),Char('d'),Char('e'))
Seq(Bol,Char('a'),Char('b'),Char('c'),Eol)
1
Syntax error in regexp 'abc(de' at position 6: Missing ')'
Syntax error in regexp 'abc[de' at position 6: Missing ']'
Syntax error in regexp 'abc)de' at position 3: Unexpected ')'
